### PR TITLE
wrappers: correctly deal with activated services during service restart

### PR DESCRIPTION
**Pull request by Meulengracht**
LP bug that triggered this fix: https://bugs.launchpad.net/snapd/+bug/2028141

The issue was that the restart code was not dealing correctly with activated services. Instead of restarting the activator units (i.e sockets, timers) for an activated service, it would just go ahead and restart the primary unit. This would start otherwise unwanted services that previously was not running (but reported enabled to the code as the activator units are enabled).

Cover this case in unit tests and correctly deal with activated services.

### CODE REVIEW

File: wrappers/services.go [hunk 1]
```diff
@@ -1223,32 +1223,64 @@ type RestartServicesOptions struct {
...
-				continue
+		var unitsToRestart []string
+
+		// If the service is activated, then we must also consider it's activators
+		if len(st.ActivatorUnitStatuses()) != 0 {
+			// If the primary unit was running then we restart it, for the primary
+			// unit we cannot take enabled into account as that is static for activated
+			// services.
+			if unitActive {
+				unitsToRestart = append(unitsToRestart, unitName)
```
Comment (bboozzoo) [inline wrappers/services.go:1260]: should this maybe happen after activators are restarted?

Comment (Meulengracht) [reply wrappers/services.go:1260]: good point, this would also be in line with the behaviour of other functions

**Review by pedronis — APPROVED** [review]
thanks

**Review by bboozzoo — APPROVED** [review]
Thanks

**Pull request merged**